app/components/views/GetStartedPage/DaemonLoading/index.js

```diff
@@ -1,38 +1,49 @@
 import React, { Component } from "react";
 import { autobind } from "core-decorators";
+import ReactTimeout from "react-timeout";
 import {
   DaemonLoadingFormHeader as DaemonLoadingHeader,
   DaemonLoadingFormBody
 } from "./Form";
 
 @autobind
-class DaemonLoadingBody extends Component {
+class DaemonLoadingBodyBase extends Component {
   constructor(props)  {
     super(props);
     this.state = this.getInitialState();
   }
 
+  getInitialState() {
+    return {
+      showLongWaitMessage: false,
+      neededBlocksDeterminedAt: new Date(),
+    };
+  }
+
   componentWillUnmount() {
     this.mounted = false;
-    this.resetState();
-    if(this.timeoutId) {
-      clearTimeout(this.timeoutId);
-    }
   }
 
-  getInitialState() {
-    return {
-      showLongWaitMessage: false
-    };
+  componentDidMount() {
+    this.mounted = true;
+    this.timeoutId = this.props.setTimeout(() => {
+      if (this.mounted) {
+        this.setState({ showLongWaitMessage: true });
+      }
+    }, 2000);
+    const neededBlocksInterval = this.props.network === "mainnet"
+      ? 5 * 60 * 1000
+      : 2 * 60 * 1000;
+    this.props.setInterval(this.props.determineNeededBlocks, neededBlocksInterval);
```

Comment:
Can you explain why you set and interval here based on network target block time?  Not sure it's necessary.  Really just need it once on startup

Reply:
During initial block download (on a first time run with a clean block database) the initial sync may take minutes/hours. During that time the block height will change, so until the daemon is completely synced to the network it should try and find the needed block height.

I understand that the difference may be small (I added the interval check mainly for completeness), so if you feel the block difference is small we can take it off.

Reply:
oh very nice!  didn't think about that situation.  nope that's totally understandable now.  nice work!